From c941ee0f80999efc6fb9fd1ad098f83fe71b476a Mon Sep 17 00:00:00 2001
From: Andreas Galauner <andreas@galauner.de>
Date: Mon, 25 Nov 2019 00:13:04 +0100
Subject: [PATCH 1/5] Fix cadence SPI driver to not use _relaxed read and write
 variants as this lead to caching problems when accessing the SPI hardware

---
 drivers/spi/spi-cadence.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-cadence.c b/drivers/spi/spi-cadence.c
index 94cc0a152449..1ca9e24ddc9d 100644
--- a/drivers/spi/spi-cadence.c
+++ b/drivers/spi/spi-cadence.c
@@ -135,12 +135,12 @@ struct cdns_spi_device_data {
 /* Macros for the SPI controller read/write */
 static inline u32 cdns_spi_read(struct cdns_spi *xspi, u32 offset)
 {
-	return readl_relaxed(xspi->regs + offset);
+	return readl(xspi->regs + offset);
 }
 
 static inline void cdns_spi_write(struct cdns_spi *xspi, u32 offset, u32 val)
 {
-	writel_relaxed(val, xspi->regs + offset);
+	writel(val, xspi->regs + offset);
 }
 
 /**
-- 
2.24.1

